core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java

```diff
@@ -485,36 +486,33 @@ private ProxiedAddresses determineProxiedAddresses(Channel channel, RequestHeade
         }
     }
 
-    private void handleOptions(ChannelHandlerContext ctx, ServiceRequestContext reqCtx) {
-        respond(ctx, reqCtx,
-                ResponseHeaders.builder(HttpStatus.OK)
-                               .add(HttpHeaderNames.ALLOW, ALLOWED_METHODS_STRING),
-                HttpData.empty(), null);
-    }
+    private InetSocketAddress remoteAddress(Channel ch) {
+        final InetSocketAddress remoteAddress = this.remoteAddress;
+        if (remoteAddress != null) {
+            return remoteAddress;
+        }
 
-    private void rejectInvalidPath(ChannelHandlerContext ctx, ServiceRequestContext reqCtx) {
-        // Reject requests without a valid path.
-        respond(ctx, reqCtx,
-                HttpStatus.BAD_REQUEST, DATA_INVALID_REQUEST_PATH,
-                new ProtocolViolationException(MSG_INVALID_REQUEST_PATH));
+        final InetSocketAddress newRemoteAddress = ChannelUtil.remoteAddress(ch);
+        this.remoteAddress = newRemoteAddress;
+        return firstNonNull(newRemoteAddress, UNKNOWN_ADDR);
```

Comment:
I am confused to understand that a nullable field is set to the field and a nonnull value is returned. Could we fall back to `UNKNOWN_ADDR` in the caller site for SRP?
It looks more verbose though. 
```java
final InetSocketAddress remoteAddress = firstNonNull(remoteAddress(channel), UNKNOWN_ADDR);

```

Reply:
Good point. Fixed both `remoteAddress()` and `localAddress()`.